src/pyhf/tensor/common.py

```diff
@@ -45,9 +45,15 @@ def stitch(self, data):
     def split(self, data):
         tensorlib, _ = get_backend()
         if len(tensorlib.shape(data)) == 1:
-            return [tensorlib.gather(data, idx) for idx in self.partition_indices]
+            return [
+                tensorlib.gather(data, tensorlib.astensor(idx, dtype='int'))
```

Comment:
this can probaly go into `_precompute` since converting the indices into tensors is not dependent on `data`

Reply:
Okay, I'll look into moving this. A note, this is _only_ being done to remove the `self.astensor` call in the PyTorch backend `gather` method (https://github.com/diana-hep/pyhf/pull/582/commits/2a163e71dc448494415960633faf72a6af1c974a). So this is a bit of a tradeoff, as this now requires extra ops by all other backends but it reduces the number of times that the PyTorch backend will have to call `astensor` overall.

Reply:
Should be resolved now.